cmd/ci-secret-bootstrap/main.go

```diff
@@ -87,37 +96,65 @@ func parseOptions() options {
 	if err := fs.Parse(os.Args[1:]); err != nil {
 		logrus.WithError(err).Errorf("cannot parse args: %q", os.Args[1:])
 	}
+	if o.vaultAddr == "" {
+		o.vaultAddr = os.Getenv("VAULT_ADDR")
+	}
+	o.vaultToken = os.Getenv("VAULT_TOKEN")
 	return o
 }
 
 func (o *options) validateOptions() error {
+	var errs []error
 	level, err := logrus.ParseLevel(o.logLevel)
 	if err != nil {
-		return fmt.Errorf("invalid log level specified: %w", err)
+		errs = append(errs, fmt.Errorf("invalid log level specified: %w", err))
 	}
 	logrus.SetLevel(level)
-	if o.bwUser == "" {
-		return fmt.Errorf("--bw-user is empty")
+
+	var credentialsProviderConfigured []string
+	if (o.bwUser == "") != (o.bwPasswordPath == "") {
+		errs = append(errs, fmt.Errorf("--bw-user and --bw-password-path must be specified together"))
+	} else if o.bwUser != "" {
+		credentialsProviderConfigured = append(credentialsProviderConfigured, "bitwarden")
 	}
-	if o.bwPasswordPath == "" {
-		return fmt.Errorf("--bw-password-path is empty")
+
+	if vals := sets.NewString(o.vaultAddr, o.vaultTokenFile, o.vaultPrefix); len(vals) != 1 && len(vals) != 3 {
```

Comment:
This is not doing exactly what it should as it will allow one of the values to be unset (one of the items in he set will be a `""`)

Reply:
Fixed